README.md
# GClient
E-Ag Yazılımının Uzak Makinede Kullandığı Grafik Arayün Client Yazılımı
<br/>
Uygulama Qt ortamında c++ ile yazılmıştır..
<br/>
Qt 5.11.2 versiyonu ile yazılmıştır.
<br/>
gcc 4.9 versiyon kullanılmıştır..
<br/>
Derlemek için kodu indiriniz.. kod adında bir dizinde açtığınızı varsayalım
<br/>
1-cd kod
<br/>
2-mkdir build
<br/>
3-cd build
<br/>
4-qmake ../GClient.pro
<br/>
5-make&make clean
<br/>
6-make install
<br/>
Bu işlemleri hatasız yaparsanız uygulama derlenecek ve sisteme yüklenecektir..
<br/>


debian/changelog
e-ag-x11client (2.2.0) unstable; urgency=medium

  * Initial Release

 -- Bayram KARAHAN <bayramk@gmail.com>  Sun, 27 Dec 2020 14:31:08 +0300


ayar.h
/*****************************************************************************
 *   Copyright (C) 2020 by Bayram KARAHAN                                    *
 *   <bayramk@gmail.com>                                                     *
 *                                                                           *
 *   This program is free software; you can redistribute it and/or modify    *
 *   it under the terms of the GNU General Public License as published by    *
 *   the Free Software Foundation; either version 3 of the License, or       *
 *   (at your option) any later version.                                     *
 *                                                                           *
 *   This program is distributed in the hope that it will be useful,         *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of          *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
 *   GNU General Public License for more details.                            *
 *                                                                           *
 *   You should have received a copy of the GNU General Public License       *
 *   along with this program; if not, write to the                           *
 *   Free Software Foundation, Inc.,                                         *
 *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA .          *
 *****************************************************************************/
#include<QCheckBox>
#include<QProcess>
#include<QProgressDialog>
#include<QToolButton>

#ifndef AYAR_H
#define AYAR_H

QWidget *MainWindow::ayar()
{

    QWidget *ayarPage=new QWidget();
     QFont f2( "Arial", 8, QFont::Normal);
    /*******************************************************/
    QStringList ayarlst=fileToList("/usr/share/e-ag/","e-ag-x11client.conf");
    /**********************************************************/
        if(listGetLine(ayarlst,"rootusername")!="")
        {
            QString strrootusername=listGetLine(ayarlst,"rootusername").split("|")[1];
            //qDebug()<<strwebblockstate;
            rootusername=strrootusername;
        }
        /**********************************************************/
            if(listGetLine(ayarlst,"rootpassword")!="")
            {
                QString strrootpassword=listGetLine(ayarlst,"rootpassword").split("|")[1];
                //qDebug()<<strwebblockstate;
                rootpassword=strrootpassword;
            }
   /*************************************************/
    if(listGetLine(ayarlst,"kilitstate")!="")
    {
        QString strkilitstate=listGetLine(ayarlst,"kilitstate").split("|")[1];
        //qDebug()<<strkilitstate;
        kilitstate=strkilitstate.toInt();
        if(kilitstate==true)
        {
            ekran->setWindowFlags(Qt::Tool);
            ekran->ekranKilit("Yönetici Mesajı:","Ekran Kilitlendi!");
            ekran->show();
        }


    }
/**********************************************************/
    if(listGetLine(ayarlst,"webblockstate")!="")
    {
        QString strwebblockstate=listGetLine(ayarlst,"webblockstate").split("|")[1];
        //qDebug()<<strwebblockstate;
        webblockstate=strwebblockstate.toInt();
        webblockcb->setChecked(webblockstate);
    }
    /*******************************************************/


    QLineEdit *webadres=new QLineEdit;
    webadres->setFixedSize(boy*9,boy);
    webadres->setText(user);
    webadres->setStyleSheet("background-color: #ffffff");

     /********************************************************/
    QToolButton *webadresEkleButton= new QToolButton;
    webadresEkleButton->setFixedSize(QSize(boy*4,boy));
    webadresEkleButton->setIconSize(QSize(boy,boy));
    webadresEkleButton->setStyleSheet("Text-align:center");
  //  webadresEkleButton->setIcon(QIcon(":/icons/save.svg"));
    webadresEkleButton->setAutoRaise(true);
    webadresEkleButton->setToolButtonStyle(Qt::ToolButtonTextUnderIcon);
    webadresEkleButton->setFont(f2);
    webadresEkleButton->setText("Web Listesine Ekle");

    connect(webadresEkleButton, &QPushButton::clicked, [=]() {
       /* kontrol();
        if (status==false) return;
    /**************************************************************/
          bool drmm1=false;
    /************************************************************************/
          QStringList webblocklist=fileToList("/usr/share/e-ag/","webblocklist");

              webblocklist<<webadres->text();

          listToFile("/usr/share/e-ag/",webblocklist,"webblocklist");
          for(int i=0;i<webblocklist.count();i++)
          {
              QString line=webblocklist[i];
              QStringList lst=line.split("|");
              twl->setRowCount(i+1);
              twl->setItem(i, 0, new QTableWidgetItem(lst[0]));//ip

          }
    });
/***********************************************************************/

twl->setFixedSize(QSize(boy*13+5,boy*6));
twl->setColumnCount(1);
//twl->setRowCount(0);
twl->setColumnWidth(0, 400);

twl->setHorizontalHeaderItem(0, new QTableWidgetItem("Web Adresi"));
twl->setSelectionBehavior(QAbstractItemView::SelectRows);
twl->setSelectionMode(QAbstractItemView::SingleSelection);
//connect(tw, &QTableWidget::cellClicked, this, cellClicked());
connect(twl, SIGNAL(cellDoubleClicked(int,int)),SLOT(cellDoubleClicked(int,int)));
twl->setRowCount(0);
QStringList list=fileToList("/usr/share/e-ag/","webblocklist");
for(int i=0;i<list.count();i++)
{
    QString line=list[i];
    QStringList lst=line.split("|");
    twl->setRowCount(twl->rowCount()+1);
    twl->setItem(i, 0, new QTableWidgetItem(lst[0]));//ip

}

/***************************************************************************/

QFont f1( "Arial", 8, QFont::Normal);
webblockcb->setFont(f1);
webblockcb->setChecked(webblockstate);
/// qDebug()<<"copystate1"<<copyState;
connect(webblockcb, &QCheckBox::clicked, [=]() {
    webblockstate=webblockcb->checkState();
   // qDebug()<<copyState;
});



    QToolButton *ayarKaydetButton= new QToolButton;
    ayarKaydetButton->setFixedSize(QSize(boy*3,boy*2));
    ayarKaydetButton->setIconSize(QSize(boy,boy));
    ayarKaydetButton->setStyleSheet("Text-align:center");
    ayarKaydetButton->setIcon(QIcon(":/icons/save.svg"));
    ayarKaydetButton->setAutoRaise(true);
    ayarKaydetButton->setToolButtonStyle(Qt::ToolButtonTextUnderIcon);
    ayarKaydetButton->setFont(f2);
    ayarKaydetButton->setText("Ayarları Kaydet");

    connect(ayarKaydetButton, &QPushButton::clicked, [=]() {
        /*kontrol();
        if (status==false) return;
/**************************************************************/
          bool drmm1=false;

          /***********************e-ag-x11client.conf oluşturuluyor***********************************/
          QStringList ayarlist;
          ayarlist.append("kilitstate|"+QString::number(kilitstate));
          ayarlist.append("webblockstate|"+QString::number(webblockstate));

          listToFile("/home/"+QDir::homePath().split("/")[2]+"/",ayarlist,"e-ag-x11client.conf");
          QString kmt18="chmod 777 /home/"+QDir::homePath().split("/")[2]+"/e-ag-x11client.conf";
          system(kmt18.toStdString().c_str());
          /*****************************e-ag-x11client********************************/
          komutSudoExpect("cp /home/"+QDir::homePath().split("/")[2]+"/e-ag-x11client.conf /usr/share/e-ag/",rootusername,rootpassword);
          komutSudoExpect("chmod 777 /usr/share/e-ag/e-ag-x11client.conf",rootusername,rootpassword);

          QString kmt27="rm -rf /home/"+QDir::homePath().split("/")[2]+"/e-ag-x11client.conf";
          system(kmt27.toStdString().c_str());

          /**********************web site bloglama bölümü***************************/
          webBlockAktifPasif();
           /***************************************************************************/
          QString sonuc=myMessageBox("e-ag", "\n"
                                             "\nAyarlar Kaydedildi..\n"
                                             "\n","","","tamam",QMessageBox::Information);

    });



    QCheckBox *kilitcb= new QCheckBox("Her Açılışta Ekranı Kilitle.",ayarPage);
    //QFont f1( "Arial", 8, QFont::Normal);
    kilitcb->setFont(f1);
    kilitcb->setChecked(kilitstate);
   /// qDebug()<<"copystate1"<<copyState;
    connect(kilitcb, &QCheckBox::clicked, [=]() {
        kilitstate=kilitcb->checkState();
       // qDebug()<<copyState;
});


    auto layout = new QGridLayout(ayarPage);
    layout->setContentsMargins(0, 0, 0,0);
    layout->addWidget(webadres, 2,0,1,1);
    layout->addWidget(webadresEkleButton, 2,1,1,1);

    layout->addWidget(twl, 5,0,1,1);
    layout->addWidget(webblockcb, 6,0,1,1);
   // layout->addWidget(webAyarKaydetButton, 7,0,1,1);


   // layout->addWidget(gelenKomut, 10,0,1,1);

    layout->addWidget(kilitcb,20,0,1,3);
    QHBoxLayout *layout1 = new QHBoxLayout;
    layout1->addWidget(ayarKaydetButton);
    layout->addLayout(layout1, 30, 0,1,3,Qt::AlignHCenter);


///layout->setColumnStretch(6, 255);

    return ayarPage;
}

void MainWindow::komutSudoExpect(QString komut,QString username,QString password)
{
    QStringList liste;
    liste<<"#!/usr/bin/expect";
    liste<<"spawn su - "+username;
    liste<<"expect \"Parola:\"";
    liste<<"send \""+password+"\\n\"";
    liste<<"send \""+password+"\\n\"";
    liste<<"send \"echo "+password+"|sudo -S "+komut+"\\n\"";
    liste<<"send \"exit\\n\"";
    liste<<"interact";
    listToFile("/home/"+QDir::homePath().split("/")[2]+"/",liste,"run.sh");


    QString kmt22="chmod 777 /home/"+QDir::homePath().split("/")[2]+"/run.sh";
    system(kmt22.toStdString().c_str());


    QString result="";
    QStringList arguments;
    arguments << "-c" << QString("/home/"+QDir::homePath().split("/")[2]+"/run.sh");
    QProcess process;
    process.start("/bin/bash",arguments);
    if(process.waitForFinished())
    {
        // version = process.readAll();
        //   result.chop(3);
    }

    QString kmt24="rm -rf /home/"+QDir::homePath().split("/")[2]+"/run.sh";
    system(kmt24.toStdString().c_str());

}
void MainWindow::cellDoubleClicked(int iRow, int iColumn)
{
     QString webadres= twl->item(iRow, 0)->text();
     QStringList list=fileToList("/usr/share/e-ag/","webblocklist");
/******************************************************************/
    //QMessageBox::StandardButton reply;
    // reply = QMessageBox::question(this, "Uyarı", "Bilgisayar Silinecek! Emin misiniz?",
      //                             QMessageBox::Yes|tr(QMessageBox::No);
     QMessageBox messageBox(this);
     messageBox.setText("Uyarı");
     messageBox.setInformativeText("Web Adresi İçin İşlem Seçiniz!");
     QAbstractButton *evetButton =messageBox.addButton(tr("Sil"), QMessageBox::ActionRole);
     QAbstractButton *hayirButton =messageBox.addButton(tr("Düzenle"), QMessageBox::ActionRole);
     messageBox.setIcon(QMessageBox::Question);
             messageBox.exec();
             if (messageBox.clickedButton() == evetButton) {
                 // qDebug()<<"evet basıldı";
                 list=listRemove(list,webadres);
                // komutSudoExpect("sudo iptables -I OUTPUT -m tcp -p tcp -d "+webadres+" --dport 443 -j ACCEPT",rootusername,rootpassword);
                 listToFile("/usr/share/e-ag/",list,"webblocklist");
                 if(list.count()==0) twl->setRowCount(0);
                 for(int i=0;i<list.count();i++)
                 {
                     QString line=list[i];
                     QStringList lst=line.split("|");
                     twl->setRowCount(i+1);
                     twl->setItem(i, 0, new QTableWidgetItem(lst[0]));//ip
                 }

                  webBlockAktifPasif();//// burası çok önemli
             }
             if (messageBox.clickedButton() == hayirButton) {
                 //qDebug()<<"hayır basıldı";
             }


}

#endif // AYAR_H


mainwindow.cpp
/*****************************************************************************
 *   Copyright (C) 2020 by Bayram KARAHAN                                    *
 *   <bayramk@gmail.com>                                                     *
 *                                                                           *
 *   This program is free software; you can redistribute it and/or modify    *
 *   it under the terms of the GNU General Public License as published by    *
 *   the Free Software Foundation; either version 3 of the License, or       *
 *   (at your option) any later version.                                     *
 *                                                                           *
 *   This program is distributed in the hope that it will be useful,         *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of          *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
 *   GNU General Public License for more details.                            *
 *                                                                           *
 *   You should have received a copy of the GNU General Public License       *
 *   along with this program; if not, write to the                           *
 *   Free Software Foundation, Inc.,                                         *
 *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA .          *
 *****************************************************************************/

#include "mainwindow.h"
#include<QTabWidget>
#include<QLabel>
#include<QGridLayout>
#include<QPushButton>
#include<QFile>
#include<QFileDialog>
#include<QRegularExpression>
#include<filecrud.h>
#include<QTimeEdit>
#include<QDate>
#include<QTimer>
#include<QTime>
#include<QMessageBox>
#include<ayar.h>
#include<giris.h>
#include<hakkinda.h>
#include<QApplication>
#include<QDesktopWidget>
#include<QMenu>
#include<QCloseEvent>
#include<QProcess>
#include<QAuthenticator>
#include<QApplication>
#include<ekranresmi.h>
#include <stdio.h>
#include <QtCore/QCoreApplication>
#include <QtDBus/QtDBus>


MainWindow::MainWindow(QWidget *parent) :
    QMainWindow(parent)
 {
    udpSocketSendMyDisp = new QUdpSocket();
    udpSocketGetX11Mesaj=new QUdpSocket();;
    udpSocketGetX11Mesaj->bind(5556, QUdpSocket::ShareAddress);
    QObject::connect(udpSocketGetX11Mesaj,&QUdpSocket::readyRead,[&](){udpSocketGetX11MesajRead();});

    /*******************************************************************/
    webblockcb= new QCheckBox("Her Açılışta Web Sitelerini Engelle.");
    twl=new QTableWidget;

    ekran=new Ekran();
    gelenKomut=new QLabel("-------------------");
    localDir="/tmp/";
    trayIcon=new QSystemTrayIcon(this);
    this->resize(340,300);
    setFixedWidth(400);
    setFixedHeight(400);
    setWindowTitle("e-ag-x11client");
    QRect screenGeometry = QApplication::desktop()->screenGeometry();
    int x = (screenGeometry.width()/2 - this->width()/2);
    int y = (screenGeometry.height() - this->height()) / 2;
    this->move(x, y);
    this->setStyleSheet("background-color: #dfdfdf;");


      //*******************tray***********************************/
      // Tray icon menu
      auto menu = this->createMenu();
      this->trayIcon->setContextMenu(menu);

      // App icon
      auto appIcon = QIcon(":/icons/e-ag-x11client.svg");
      this->trayIcon->setIcon(appIcon);
      this->setWindowIcon(appIcon);

      // Displaying the tray icon
      this->trayIcon->show();     // Note: without explicitly calling show(), QSystemTrayIcon::activated signal will never be emitted!

      // Interaction
      connect(trayIcon, &QSystemTrayIcon::activated, this, &MainWindow::iconActivated);
     // gizle();

      timergizle = new QTimer(this);
      connect(timergizle, SIGNAL(timeout()), this, SLOT(gizle()));
      timergizle->start(1);
      aw=ayar();

      aw->setEnabled(false);
       /************************************************************/
        tw=new QTabWidget(this);
        tw->resize(this->width(),this->height());
        tw->addTab(giris(),"Giriş");
        tw->addTab(aw,"Ayarlar");
        tw->addTab(hakkinda(),"Hakkında");


/***************************************************************/
        webBlockAktifPasif();
/***************************************************************/
    QStringList arguments;
        arguments << "-c" << QString("printenv USER");
        QProcess process;
        process.start("/bin/bash",arguments);
        if(process.waitForFinished())
        {
            x11user = process.readAll();
           /// qDebug()<<"mydisp user bilgi:"<<user;
               x11user.chop(1);
        }
        QStringList argumentss;
        argumentss << "-c" << QString("printenv DISPLAY");


        process.start("/bin/bash",argumentss);
        if(process.waitForFinished())
        {
            x11display = process.readAll();
            ///qDebug()<<"mydisp display bilgi:"<<display;
               x11display.chop(1);
        }


        if(!x11display.contains("0", Qt::CaseInsensitive))//!=0
        {
             QString kmt20="nohup /usr/bin/x11vnc -forever -loop -noxdamage -repeat -rfbauth /usr/bin/x11vncpasswd -rfbport 5901 -shared &";
             system(kmt20.toStdString().c_str());
        }

 /**************************************************************************/
        QTimer *mydispTimer = new QTimer();
        QObject::connect(mydispTimer, &QTimer::timeout, [&](){

            x11mydispresult=x11user+"|"+
                    x11display+"|"+
                    QString::number(kilitstate)+"|"+
                    QString::number(transparankilitstate)+"|"+
                    QString::number(ekranimagestate);
               QByteArray datagram = x11mydispresult.toUtf8();// +QHostAddress::LocalHost;
            udpSocketSendMyDisp->writeDatagram(datagram,QHostAddress::LocalHost, 5555);
            ////qDebug()<<"server'a mesaj gönderildi:"<< serverlist[0]<<_data;
        });
        mydispTimer->start(2000);

        /***************************************************************************/


       }

void MainWindow::udpSocketGetX11MesajRead()
{
    QByteArray datagram;

    while (udpSocketGetX11Mesaj->hasPendingDatagrams()) {
        datagram.resize(int(udpSocketGetX11Mesaj->pendingDatagramSize()));
        QHostAddress sender;
        quint16 senderPort;

        udpSocketGetX11Mesaj->readDatagram(datagram.data(), datagram.size(), &sender, &senderPort);

        QString rmesaj=datagram.constData();
     ///   qDebug()<<"x11'den Gelen Udp Mesaj:"<<rmesaj;
   tcpMessageControlSlot(rmesaj);

    }
}
void MainWindow::webBlockAktifPasif()
{
    QStringList ayarlst=fileToList("/usr/share/e-ag/","e-ag-x11client.conf");


    /**********************************************************/
        if(listGetLine(ayarlst,"rootusername")!="")
        {
            QString strrootusername=listGetLine(ayarlst,"rootusername").split("|")[1];
            //qDebug()<<strwebblockstate;
            rootusername=strrootusername;
        }
        /**********************************************************/
            if(listGetLine(ayarlst,"rootpassword")!="")
            {
                QString strrootpassword=listGetLine(ayarlst,"rootpassword").split("|")[1];
                //qDebug()<<strwebblockstate;
                rootpassword=strrootpassword;
            }

/**********************************************************/
    if(listGetLine(ayarlst,"webblockstate")!="")
    {
        QString strwebblockstate=listGetLine(ayarlst,"webblockstate").split("|")[1];
        //qDebug()<<strwebblockstate;
        webblockstate=strwebblockstate.toInt();
        webblockcb->setChecked(webblockstate);
    }
    /******************************************************************/
    if(webblockstate==true)
    {
     komutSudoExpect("/usr/share/e-ag/webdisable.sh",rootusername,rootpassword);
    }
    if(webblockstate==false)
    {
    komutSudoExpect("iptables -F",rootusername,rootpassword);
    }
       /*********************************************************/

}
void MainWindow::tcpMessageControlSlot(QString _data)
{

    QDesktopWidget widget;
    QRect desktopScreenSize = widget.availableGeometry(widget.primaryScreen());
    FileCrud *fcc=new FileCrud();
    fcc->dosya=localDir+"tcpMessage";
    ///qDebug()<<"dosya bilgi:"<<fcc->dosya;
    /**************************************************************************/
    if (ekranimagestate)
    {
       kaydetTimerSlot(rootusername,rootpassword);
    }
  

  //  if(fcc->fileExists()&&fileToList(localDir,"tcpMessage").count()!=0)
   // {
   //     QStringList str=fileToList(localDir,"tcpMessage");
    QStringList str;
    str.append(_data);
        if(str[0].count()==0){
          ////  qDebug()<<"gelen bilgi YOK!!";
            return;}
        qDebug()<<"gelen bilgi:"<<str;
        QStringList lst=str[0].split("|");
        gelenKomut->setText(lst[1]+"  "+lst[2]);


        if(lst[0]=="dosyatopla")
        {

            //lst[2]
            QString severip=lst[2];
            //lst[4]
            QString username=lst[4];
            //lst[5]
            QString password=lst[5] ;
            QString localusername=lst[5] ;
            hostAddressMacButtonSlot();//local ip adresi tespit ediliyor.

            QDir directory(QDir::homePath()+"\/Masaüstü");
             QStringList filelist = directory.entryList(QStringList() << "e-ag-server.*",QDir::Files);
             QString ad="";
             QString gercekad="";
             foreach(QString filename, filelist) {
                 QFileInfo fi(filename);
                 QString uzanti = fi.completeSuffix();
                 gercekad=QDir::homePath()+"\/Masaüstü\/"+filename;
                 ad="-e-ag-server."+uzanti;
             }

             for(int i=0;i<ipmaclist.count();i++)
             {
                 QString komut="sshpass -p "+password+" scp -o StrictHostKeyChecking=no "+gercekad+" "+
                         username+"@"+severip+":\/home\/"+username+"\/Masaüstü\/"+ipmaclist[i].ip+ad;

                 qDebug()<<komut;

                 QStringList arguments;
                 arguments << "-c" << komut;
                 QProcess process;
                 process.start("/bin/bash",arguments);
                 process.waitForFinished(-1); // will wait forever until finished
             }

        }
        if(lst[0]=="webblock")
        {
           

            webblockstate=lst[1].toInt();
            webblockcb->setChecked(webblockstate);

            twl->setRowCount(0);
            QStringList list=fileToList("/usr/share/e-ag/","webblocklist");
            for(int i=0;i<list.count();i++)
            {
                QString line=list[i];
                QStringList lst=line.split("|");
                twl->setRowCount(twl->rowCount()+1);
                twl->setItem(i, 0, new QTableWidgetItem(lst[0]));//ip

            }

            /*************************************************/
            webBlockAktifPasif();
            /**************************************************/

        }
        if(lst[0]=="ekranmesaj")
        {
            qDebug()<<"ekranmesaj:"<<lst[0]<<lst[1];
            ekran->setWindowFlags(Qt::Tool);
            ekran->ekranMesaj("Yönetici Mesajı:",lst[1]);
            ekran->show();

        }
        if(lst[0]=="kilitstatetrue")
        {
            qDebug()<<"ekrankilittrue:"<<lst[0];
            //ekran->close();
            if (!kilitstate)
            {
                kilitstate=true;
                transparankilitstate=false;
                ekranimagestate=false;
                ekran->setWindowFlags(Qt::Tool);
                ekran->ekranKilit("Yönetici Mesajı:","Ekran Kilitlendi!");
                ekran->show();
            }

        }
        if(lst[0]=="kilitstatefalse")
        {
            qDebug()<<"ekrankilitac:"<<lst[0];
            ekran->close();
            kilitstate=false;
        }
        if(lst[0]=="transparankilitstatetrue")
        {
            qDebug()<<"kilittransparanstatetrue";
            if (!transparankilitstate)
            {
                kilitstate=false;
                transparankilitstate=true;
                ekranimagestate=false;
                ekran->setWindowFlags(Qt::Tool);
                ekran->ekranTransparanKilit();
                ekran->show();
            }

        }
        if(lst[0]=="transparankilitstatefalse")
        {
            qDebug()<<"ekrantransparankilitac:"<<lst[0];
            ekran->close();
            transparankilitstate=false;
        }
        if(lst[0]=="ekranimagestatetrue")
        {
            qDebug()<<"GClient:"<<lst[0];
            if (!ekranimagestate)
            {
                kilitstate=false;
                transparankilitstate=false;
                ekranimagestate=true;

                kaydetTimerSlot(lst[4],lst[5]);
                rootusername=lst[4];
                rootpassword=lst[5];

            }
        }
        if(lst[0]=="ekranimagestatefalse")
        {
            qDebug()<<"GClient:"<<lst[0];
           ekranimagestate=false;
            kaydetTimerSlot(lst[4],lst[5]);

        }
        if(lst[0]=="x11komut")
        {
           // qDebug()<<"komut:"<<lst[1]<<lst[2]<<lst[3]<<lst[4]<<lst[5]<<lst[6]<<lst[7];
           QString komut="nohup "+lst[1]+" &";
            system(komut.toStdString().c_str());
          //  komutSudoExpect(lst[1],lst[6],lst[7]);
        }
  /*      QString kmt21="echo "" >/tmp/tcpMessage";
        system(kmt21.toStdString().c_str());
    }
        else{
            qDebug()<<"tcpMessage Dosyası Yok!!";
            return;
            ///exit(0);
        }*/
   }
QString MainWindow::myMessageBox(QString baslik, QString mesaj, QString evet, QString hayir, QString tamam, QMessageBox::Icon icon)
{
    Qt::WindowFlags flags = 0;
    flags |= Qt::Dialog;
    flags |= Qt::X11BypassWindowManagerHint;

    QMessageBox messageBox(this);
    messageBox.setWindowFlags(flags);
    messageBox.setText(baslik+"\t\t\t");
    messageBox.setInformativeText(mesaj);
    QAbstractButton *evetButton;
    QAbstractButton *hayirButton;
    QAbstractButton *tamamButton;

    if(evet=="evet") evetButton =messageBox.addButton(tr("Evet"), QMessageBox::ActionRole);
    if(hayir=="hayir") hayirButton =messageBox.addButton(tr("Hayır"), QMessageBox::ActionRole);
    if(tamam=="tamam") tamamButton =messageBox.addButton(tr("Tamam"), QMessageBox::ActionRole);

    messageBox.setIcon(icon);
    messageBox.exec();
    if(messageBox.clickedButton()==evetButton) return "evet";
    if(messageBox.clickedButton()==hayirButton) return "hayır";
    if(messageBox.clickedButton()==tamamButton) return "tamam";
    return "";
}

MainWindow::~MainWindow()
{
  //  delete ui;
}

void MainWindow::hostAddressMacButtonSlot()
{
    QHostAddress localhost = QHostAddress(QHostAddress::LocalHost);
ipmaclist.clear();
    foreach (const QNetworkInterface& networkInterface, QNetworkInterface::allInterfaces()) {
           foreach (const QNetworkAddressEntry& entry, networkInterface.addressEntries()) {
               QHostAddress *hostadres=new QHostAddress(entry.ip().toString());
               if(hostadres->protocol() == QAbstractSocket::IPv4Protocol &&
                       !hostadres->isLoopback() )
               {
                  IpMac im;
                  im.ip=entry.ip().toString();
                  im.mac=networkInterface.hardwareAddress();
                  im.broadcast=entry.broadcast().toString();
                  im.subnet=entry.netmask().toString();
                  ipmaclist.append(im);
                 // qDebug()<<"mac:"<<networkInterface.hardwareAddress();
                  //qDebug()<<"ip  address:"<<entry.ip().toString();
                 // qDebug()<<"broadcast  address:"<<entry.broadcast().toString();
              }
           }
       }



}

QStringList MainWindow::listRemove(QStringList list,QString data)
 {
       QRegularExpression re(data);
     for(int i=0;i<list.count();i++)if(list[i].contains(re)) list.removeAt(i);
    // qDebug()<<list;
     return list;
 }
QString MainWindow::listGetLine(QStringList list,QString data)
{
    QRegularExpression re(data);
   for(int i=0;i<list.count();i++) if(list[i].contains(re)) return list[i];
   //qDebug()<<list;
   return "";
}
QStringList MainWindow::fileToList(QString path, QString filename)
{
  FileCrud *fcc=new FileCrud();
  fcc->dosya=path+filename;
 // qDebug()<<fcc->dosya;
  QStringList list;
  for(int i=1;i<=fcc->fileCount();i++)
  {
       QString line=fcc->fileGetLine(i);
      // qDebug()<<line;
       if(line!="")
       {
           line.chop(1);
           QStringList lst=line.split("|");
           QString ln="";
           if(lst.count()>0)ln.append(lst[0]);
           if(lst.count()>1)ln.append("|").append(lst[1]);
           if(lst.count()>2)ln.append("|").append(lst[2]);
           if(lst.count()>3)ln.append("|").append(lst[3]);
           if(lst.count()>4)ln.append("|").append(lst[4]);
           if(lst.count()>5)ln.append("|").append(lst[5]);
           if(lst.count()>6)ln.append("|").append(lst[6]);
           if(lst.count()>7)ln.append("|").append(lst[7]);
           if(lst.count()>8)ln.append("|").append(lst[8]);
           if(lst.count()>9)ln.append("|").append(lst[9]);
           if(lst.count()>10)ln.append("|").append(lst[10]);
           if(lst.count()>11)ln.append("|").append(lst[11]);
           if(lst.count()>12)ln.append("|").append(lst[12]);
           if(lst.count()>13)ln.append("|").append(lst[13]);

             list <<ln;
           //qDebug()<<ln;
           // list <<lst[0]+"|"+lst[1]+"|"+lst[2]+"|"+lst[3]+"|"+lst[4]+"|"+lst[5];

       }
  }
     return list;
}
void MainWindow::listToFile(QString path,QStringList list, QString filename)
{
  FileCrud *fcc=new FileCrud();
  fcc->dosya=path+filename;
  fcc->fileRemove();
  for(int i=0;i<list.count();i++)
  {
       QString line=list[i];
       if(line!="")
       {
           //line.chop(1);
           QStringList lst=line.split("|");
           //qDebug()<<line;
           QString ln="";
           if(lst.count()>0)ln.append(lst[0]);
           if(lst.count()>1)ln.append("|").append(lst[1]);
           if(lst.count()>2)ln.append("|").append(lst[2]);
           if(lst.count()>3)ln.append("|").append(lst[3]);
           if(lst.count()>4)ln.append("|").append(lst[4]);
           if(lst.count()>5)ln.append("|").append(lst[5]);
           if(lst.count()>6)ln.append("|").append(lst[6]);
           if(lst.count()>7)ln.append("|").append(lst[7]);
           if(lst.count()>8)ln.append("|").append(lst[8]);
           if(lst.count()>9)ln.append("|").append(lst[9]);
           if(lst.count()>10)ln.append("|").append(lst[10]);
           if(lst.count()>11)ln.append("|").append(lst[11]);
           if(lst.count()>12)ln.append("|").append(lst[12]);
           if(lst.count()>13)ln.append("|").append(lst[13]);

           //qDebug()<<ln;
           fcc->fileWrite(ln);

       }

  }
}


void  MainWindow::gizle()
{
    //hide();
    aw->setEnabled(false);
    tw->setCurrentIndex(0);
    QWidget::hide();
    timergizle->stop();

  }
void  MainWindow::widgetShow()
{
    if(passwordKontrolSlot("pkexec sh -c 'touch /tmp/eagx11client'"))
    {
      //  aw->setEnabled(true);
       // tw->setCurrentIndex(1);
        this->showNormal();
        twl->setRowCount(0);
        QStringList list=fileToList("/usr/share/e-ag/","webblocklist");
        for(int i=0;i<list.count();i++)
        {
            QString line=list[i];
            QStringList lst=line.split("|");
            twl->setRowCount(twl->rowCount()+1);
            twl->setItem(i, 0, new QTableWidgetItem(lst[0]));//ip
        }
        /****************************************************************/
        aw->setEnabled(true);
        tw->setCurrentIndex(1);
    }


 }
bool MainWindow::passwordKontrolSlot(QString kmt)
{
    system(kmt.toStdString().c_str());
   // qDebug()<<"denememe";
    if(QFile::exists("/tmp/eagx11client")==true)
    {
        //system("rm -rf /tmp/eagx11client");
        qDebug()<<"Şifre: doğru ";
        return true;
    }else
    {
       qDebug()<<"Şifre: yanlış ";
        return false;
    }
return false;
}

QMenu* MainWindow::createMenu()
{

    auto minimizeAction = new QAction(tr("Gi&zle"), this);
    connect(minimizeAction, &QAction::triggered, this, &QWidget::hide);
    auto restoreAction = new QAction(tr("&Ayarlar"), this);
    connect( restoreAction, SIGNAL(triggered()), this, SLOT(widgetShow()));
    auto menu = new QMenu(this);
    menu->addAction(minimizeAction);
    //menu->addAction(maximizeAction);
    menu->addAction(restoreAction);
    menu->addSeparator();

    return menu;
}

void MainWindow::iconActivated(QSystemTrayIcon::ActivationReason reason_)
{
  switch (reason_) {
  case QSystemTrayIcon::Trigger:
   // this->trayIcon->showMessage("e-ag", "İstemci Uygulaması!");
        widgetShow();
    break;
  default:
    ;
  }
}

void MainWindow::WidgetClosed()
{
    aw->setEnabled(false);
    tw->setCurrentIndex(0);
    QWidget::hide();
}

void MainWindow::closeEvent(QCloseEvent *event)
{
     emit WidgetClosed();
     //QWidget::hide();
     event->ignore();

}
